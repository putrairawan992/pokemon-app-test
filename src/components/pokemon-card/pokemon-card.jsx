import React from "react";
import T from "prop-types";
import { makeStyles } from "@material-ui/core/styles";
import Card from "../card/card";
import CardActionArea from "@material-ui/core/CardActionArea";
import CardContent from "@material-ui/core/CardContent";
import CardMedia from "@material-ui/core/CardMedia";
import Typography from "@material-ui/core/Typography";
import Link from "@material-ui/core/Link";

const propTypes = {
  image: T.string,
  name: T.string,
  types: T.arrayOf(T.string),
  abilities: T.arrayOf(T.string),
};

const useStyles = makeStyles({
  root: {
    maxWidth: 345,
  },
  media: {
    height: 140,
  },
});

export const PokemonCard = ({ image, name, types, abilities }) => {
  const classes = useStyles();

  return (
    <Card className={classes.root}>
      <CardActionArea>
        <Link href={image}>
          <CardMedia className={classes.media} image={image} title={name} />
        </Link>
        <CardContent>
          <Typography gutterBottom variant="h5" component="h2">
            {name}
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p">
            Types: {types.join(", ")}
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p">
            Abilities: {abilities.join(", ")}
          </Typography>
        </CardContent>
      </CardActionArea>
    </Card>
  );
};

PokemonCard.propTypes = propTypes;
