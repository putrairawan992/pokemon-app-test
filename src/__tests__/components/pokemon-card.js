import React from "react";
import renderer from "react-test-renderer";
import { PokemonCard } from "../../components/pokemon-card/pokemon-card";

it("Should render pokemon card correctly", () => {
  const component = renderer.create(
    <PokemonCard
      image=""
      name="bulbasaur"
      types={["grass"]}
      abilities={["move1", "move2"]}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
