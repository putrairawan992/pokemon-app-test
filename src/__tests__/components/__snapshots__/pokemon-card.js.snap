// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render pokemon card correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <a
      className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      href=""
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        className="MuiCardMedia-root makeStyles-media-2"
        title="bulbasaur"
      />
    </a>
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        bulbasaur
      </h2>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Types: 
        grass
      </p>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Abilities: 
        move1, move2
      </p>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight"
    />
  </button>
</div>
`;
